compose/utils.py

```diff
@@ -170,3 +170,12 @@ def truncate_id(value):
     if len(value) > 12:
         return value[:12]
     return value
+
+
+def unique_everseen(iterable, key=lambda x: x):
+    "List unique elements, preserving order. Remember all elements ever seen."
+    seen = set()
+    for element in iterable:
+        if key(element) not in seen:
```

Comment:
You're testing for the presence of `key(element)` in the `set` but then add `element` to it, which won't work:
```
In [7]: def keyf(x):
   ...:     return 'key:' + x

In [8]: list(unique_everseen(['a', 'b', 'c', 'a'], keyf))
Out[8]: ['a', 'b', 'c', 'a']
```

Reply:
Oups, you are right. Fixing that.

I had to adapt for it to be python2.7-compliant and I missed that. 
I'll update the tests too.